File: Python/program/course.py
# Saya Muhamad Firdaus [2101995] mengerjakan
# soal Latihan 4 dalam mata kuliah DPBO
# untuk keberkahanNya maka saya tidak melakukan kecurangan 
# seperti yang telah dispesifikasikan. Aamiin.

class Course():
    # Constructor
    def __init__(self, nama_matakuliah):
        self.nama_matakuliah = nama_matakuliah
        self.Mhsiswa = []
        self.Dos = []
        
    # Setter dan Getter
    def setNamaMataKuliah(self, nama_matakuliah):
        self.nama_matakuliah = nama_matakuliah
        
    def getNamaMataKuliah(self):
        return self.nama_matakuliah
    
    def getMhsiswa(self):
        return self.Mhsiswa
    
    def getDos(self):
        return self.Dos
    
    # Tambah mahasiswa
    def addMahasiswa(self, mahasiswa):
        self.Mhsiswa.append(mahasiswa)
    
    # Tambah dosen
    def addDosen(self, dosen):
        self.Dos.append(dosen)

File: cpp/program/course.cpp
// Saya Muhamad Firdaus [2101995] mengerjakan
// soal Latihan 4 dalam mata kuliah DPBO
// untuk keberkahanNya maka saya tidak melakukan kecurangan 
// seperti yang telah dispesifikasikan. Aamiin.

#include "header.hh"

// Constructor
Course::Course(){
    this->nama_matakuliah = "";
}

// Constructor
Course::Course(string nama_matakuliah){
    this->nama_matakuliah = nama_matakuliah;
}
        
// Setter dan Getter
void Course::setNamaMataKuliah(string nama_matakuliah){
    this->nama_matakuliah = nama_matakuliah;
}
        
string Course::getNamaMataKuliah(){
    return nama_matakuliah;
}
     
vector<Mahasiswa> Course::getMhsiswa(){
    return Mhsiswa;
}
        
vector<Dosen> Course::getDos(){
    return Dos;
}
        
// Tambah mahasiswa
void Course::addMahasiswa(Mahasiswa mahasiswa){
    Mhsiswa.push_back(mahasiswa);
}
        
// Tambah dosen
void Course::addDosen(Dosen dosen){
    Dos.push_back(dosen);
}

// Destructor
Course::~Course(){

}